wagtail/models/__init__.py

```diff
@@ -133,6 +137,9 @@
 from .specific import SpecificMixin
 from .view_restrictions import BaseViewRestriction
 
+if TYPE_CHECKING:
+    from django.http import HttpRequest
```

Comment:
This is restoring the imports that existed prior to 88a3a397ca6e756ee644d87fdac51ba81cc971c1 (which made the import of HttpRequest unconditional).